realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -112,6 +112,7 @@ public boolean handleMessage(Message message) {
                 }
                 case COMPLETED_UPDATE_ASYNC_QUERIES: {
                     // this is called once the background thread completed the update of the async queries
+                    notifyAllListeners();
```

Comment:
Is this correct? `completedAsyncQueriesUpdate` will emit `REALM_CHANGED` again, and `notifyAllListeners()` will be called anyway.